### ISSUE TITLE
Feature request: Controller leader election

### ISSUE THREAD

**Issue by z0rc** [STRRL/cloudflare-tunnel-ingress-controller#130]
It's expected to run multiple controller replicas simultaneously for HA. In the same time, given the nature of controller's logic, it doesn't make sense to allow every replica do the actuall work, as this can trigger race condition.

https://sklar.rocks/kubernetes-leader-election/

**Comment by STRRL** [comment]
Yeah, that would be coll and could be easily supported with controller-runtime, https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/leaderelection . 

I am very appriciate for your great idea for pushing this project near to production available step by step! 

Do you have interests in building it? ❤️ Or I could make it in the next days. 🚀

**Comment by z0rc** [comment]
@STRRL my Go skills are at that level, so it's up to you. Thanks.

**Comment by clementblaise** [comment]
With #70 leader election is not needed

**Comment by STRRL** [comment]
Done in #272: the controller supports `--leader-elect` (wired to controller-runtime leader election), the Helm chart exposes `leaderElection.enabled`, and the connector reconcile loop is gated on election. Multiple replicas can run safely for HA.
